service/controller/cluster.go

```diff
@@ -32,9 +34,11 @@ type ClusterConfig struct {
 	CertTTL                    string
 	ClusterIPRange             string
 	DNSIP                      string
+	Flag                       *flag.Flag
```

Comment:
We get all the settings from viper in `service.go`.  So just add new settings to the wiring for the new config.


https://github.com/giantswarm/cluster-operator/blob/7219d70e4a61a98564e27a901ebd57a19635a5ac/service/service.go

Reply:
Would that makes sense to get raw config strings and marshal only from this resource level?

Reply:
I don't think we will hit any limits on string length. We don't have that many apps.

I think it's cleaner to get the settings in service.go. Instead of passing the flag and viper objects all the way to the resource.